crates/rpc-types-beacon/src/requests.rs

```diff
@@ -0,0 +1,181 @@
+};
+use serde::{Deserialize, Serialize};
+
+/// An Electra-compatible execution requests payload.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "ssz", derive(ssz_derive::Decode, ssz_derive::Encode))]
+pub struct ExecutionRequestsV4 {
+    /// The requested deposits.
+    pub deposits: Vec<DepositRequest>,
+    /// The requested withdrawals.
+    pub withdrawals: Vec<WithdrawalRequest>,
+    /// The requested consolidations.
+    pub consolidations: Vec<ConsolidationRequest>,
+}
+
+#[cfg(feature = "ssz")]
+pub use ssz_requests_conversions::TryFromRequestsError;
+
+#[cfg(feature = "ssz")]
+mod ssz_requests_conversions {
+    use super::*;
+    use crate::requests::TryFromRequestsError::SszDecodeError;
+    use alloy_eips::{
+        eip6110::{DepositRequest, DEPOSIT_REQUEST_TYPE, MAX_DEPOSIT_RECEIPTS_PER_PAYLOAD},
+        eip7002::{WithdrawalRequest, MAX_WITHDRAWAL_REQUESTS_PER_BLOCK, WITHDRAWAL_REQUEST_TYPE},
+        eip7251::{
+            ConsolidationRequest, CONSOLIDATION_REQUEST_TYPE, MAX_CONSOLIDATION_REQUESTS_PER_BLOCK,
+        },
+        eip7685::Requests,
+    };
+    use ssz::{Decode, DecodeError, Encode};
+
+    impl TryFrom<&Requests> for ExecutionRequestsV4 {
+        type Error = TryFromRequestsError;
+
+        fn try_from(value: &Requests) -> Result<Self, Self::Error> {
+            #[derive(Default)]
+            struct RequestAccumulator {
+                deposits: Vec<DepositRequest>,
+                withdrawals: Vec<WithdrawalRequest>,
+                consolidations: Vec<ConsolidationRequest>,
+            }
+
+            impl RequestAccumulator {
+                fn parse_request_payload<T>(
+                    payload: &[u8],
+                    max_size: usize,
+                    request_type: u8,
+                ) -> Result<Vec<T>, TryFromRequestsError>
+                where
+                    Vec<T>: Decode + Encode,
+                {
+                    let list: Vec<T> = Vec::from_ssz_bytes(payload)
+                        .map_err(|e| SszDecodeError(request_type, e))?;
+
+                    if list.len() > max_size {
+                        return Err(TryFromRequestsError::RequestPayloadSizeExceeded(
+                            request_type,
+                            list.len(),
+                        ));
+                    }
+
+                    Ok(list)
+                }
+
+                fn accumulate(mut self, request: &[u8]) -> Result<Self, TryFromRequestsError> {
+                    if request.is_empty() {
+                        return Err(TryFromRequestsError::EmptyRequest);
+                    }
+
+                    let (request_type, payload) =
+                        request.split_first().expect("already checked for empty");
+
+                    match *request_type {
+                        DEPOSIT_REQUEST_TYPE => {
```

Comment:
this doesn't enforce any ordering, but this is fine because this is validated via the requests hash anyway

Reply:
Ya, technically this'll accept an "out of order" `Requests` (and then reorder it if converted back).  Agree it's fine since the hashes wouldn't match, if it ends up coming up often we can add a check that the types are in order.